torch/csrc/jit/passes/quantization.cpp

```diff
@@ -1009,7 +1019,9 @@ bool InsertObserversHelper::valueNeedsToBeQuantized(Value* v) {
   }
   // Check whether user is quantizable
   for (const auto& use : v->uses()) {
-    if (nodeQuantizable(use.user) || isDynamicLSTMWeight(v, use, is_dynamic)) {
+    if ((nodeQuantizable(use.user) &&
+         nodeInputQuantizable(use.user, use.offset)) ||
+        isDynamicLSTMWeight(v, use, is_dynamic)) {
```

Comment:
I think you can add a `valueQuantizable` function and use that for `isDynamicLSTMWeight` check as well

Reply:
we probably need another refactor to unify these two actually

Reply:
@supriyar could you have a separate PR before this one that refactors `nodeQuantizable` to `valueQuantizable` that takes a `value` as input?